### ISSUE TITLE
Languages for each repo

### ISSUE THREAD

**Issue by aljesusg** [RedHatOfficial/RedHatOfficial.github.io#139]
Like opensource contributor would be great see the programming languages ​​of each repo, future contributors can search by program language to see in which projects can collaborate.

It's only add to the projects hash an array of languages and we can shown them  with some opensource icon library like [Devicon](http://konpa.github.io/devicon/) or [font-mfizz](http://fizzed.com/oss/font-mfizz) with a tooltip with the languages.

What do you think?

**Comment by aljesusg** [comment]
Hi @Fryguy ^^ nice to see you again, Yes we can pull the information my question is if I can do this improve.

**Comment by Fryguy** [comment]
@castastrophe Thoughts on this?  I like the idea, but I'm not sure how to present it in a nice way.

**Comment by frcr** [comment]
I have somewhat workable solution for this, although I came up with a few questions regarding best approach.

**tl;dr:** Should we pull languages dynamically at every launch user-side or store them in repo, updating from time to time through commits?

1) The idea of this project seems for it to be self-contained, i.e. decoupled from external sources (apart from some external css's and fonts). Pulling individual language statistics per each repo from github API seems to break this restriction.

2) Github imposes limit of 60/hour on the number of anonymous calls to its API. The number of required calls to pull language statistics far exceeds this. For authorized requests this goes up to 5000/hour, which would cover our needs, but requires a token.

3) My current solution, while solving the aforementioned problems, suffers from its own: I have a python script that pulls down all the needed info once and updates all the projects in the `./app/data/projects.json` file with an additional field `"languages"` showing their names and usage percentages per project (for multirepo projects it's percentage across all repos). This allows for perfect decoupling and removes the whole token/authorization burden from the end user. But it will require periodic updates from maintainers and whoever adds a new project to the list will have to either provide their own data or launch the script once so that it updates the newly added project with language statistics.

I updated the `dynamictable.html` template to accommodate for this new field. I'm not a designer, so it ain't pretty, but not eye-bleedingly ugly either.

**Comment by bproffitt** [comment]
Okay, so the point of this page is to highlight projects in which Red Hat participates. Adding language information is helpful, but if we go this direction, now the site becomes more utilitarian in its mission. To be clear, this is _not_ a bad thing. In fact, there is an upcoming effort from my team that will be highlighting lots of projects for ease of onboarding, so this feature could be ported to or from that project's outlet.

**Comment by FirdausJawed** [comment]
Is this issue is for adding the highlights of the projects in which red hat is participating?

can i work upon it?

can you tell me where to get started?

really interested in this project
